clash-ghc/clash-ghc.cabal

```diff
@@ -55,17 +55,27 @@ source-repository head
   type: git
   location: https://github.com/clash-lang/clash-compiler.git
 
+flag dynamic
+  description:
+    Build the clash binaries with -dynamic
+  default: True
+  manual: True
+
 executable clash
   Main-Is:            src-ghc/Batch.hs
   Build-Depends:      base, clash-ghc
   GHC-Options:        -Wall
+  if flag(dynamic)
+    GHC-Options: -dynamic
```

Comment:
No, cabal should handle dynamic binary handling using `--enable-executable-dynamic`, it has better RPATH handling than GHC. See also: https://github.com/clash-lang/clash-compiler/issues/314

Reply:
A few things:

1. The Debian build scripts use fakeroot and (probably?) won't generate bad RPATH directories the check in #314 (https://github.com/openSUSE/brp-check-suse/blob/master/brp-35-rpath) checks for.
2. The Debian build scripts don't support passing a flag such as `--enable-executable-dynamic` to cabal, it only allows boolean flags like the one above. Trying to "fix" this tool seems a waste of time if we can just use ghc-options IMO.
3. Either way, the problem in #314 is that it's a hardcoded flag. It doesn't have to be, as this patch proves.
4. We could set the default to false.
5. We could document the flag to indicate "normal" users probably don't want to touch it.
6. I'm not sure why we even care about better RPATH handling. Does it have any tangible benefits?

Reply:
What if we rename the flag to `force-dynamic` with a default state to `False`?

Reply:
I can live with a `False`  default, and mention in the flag description that his flag is only used for packaging, and that uses should use `--enable-executable-dynamic`.

Reply:
Perfect!